rafthttp/stream.go

```diff
@@ -38,6 +39,16 @@ const (
 	streamBufSize = 4096
 )
 
+var (
+	errUnsupportedStreamType = fmt.Errorf("unsupported stream type")
+
+	streamSupportMap = map[streamType]*semver.Version{
```

Comment:
If we want to deprecate one, we could introduce max version map to check that.

If remote tells me its version is 2.2, it should support all what 2.1 has today; and if remote is 2.3, it should stop talking because it is two-version behind, which will be done in enforcement.

Both ways are good to me.